protos/feast/serving/Serving.proto

```diff
@@ -26,51 +26,29 @@ option java_outer_classname = "ServingAPIProto";
 option go_package = "github.com/gojek/feast/protos/generated/go/feast/serving";
 
 service ServingAPI {
-    // Query features from Feast
-    rpc QueryFeatures (QueryFeatures.Request) returns (QueryFeatures.Response) {};
+    // Query features from Feast serving storage
+    rpc QueryFeatures (QueryFeaturesRequest) returns (QueryFeaturesResponse) {};
 }
 
-message QueryFeatures {
-    message Request {
-        // e.g. "driver", "customer", "city".
-        string entityName = 1;
-        // List of entity ID.
-        repeated string entityId = 2;
-        // List of request details, contains: featureId, type of query, and limit.
-        repeated RequestDetail requestDetails = 3;
-        // Filter specifying only to retrieve features having timestamp within this range.
-        TimestampRange timestampRange = 4;
-    }
-
-    message Response {
-        // e.g. "driver", "customer", "city".
-        string entityName = 1;
-        // map of entity ID and its entity's properties.
-        map<string, Entity> entities = 2;
-    }
-}
-
-message RequestDetail {
-    // feature ID to be included in the query.
+message QueryFeaturesRequest {
+    // e.g. "driver", "customer", "city".
+    string entityName = 1;
+    // List of entity ID.
+    repeated string entityId = 2;
+    // List of feature ID.
     // feature ID is in the form of [entity_name].[granularity].[feature_name]
     // e.g: "driver.day.total_accepted_booking"
     // all requested feature ID shall have same entity name.
-    string featureId = 1;
-    // request type either LAST or LIST.
-    // LAST : return only the latest value based on timestamp.
-    // LIST : return list of historical data sorted by timestamp.
-    RequestType type = 2;
-    // only applicable to LIST.
-    // length of the returned list <= limit.
-    // default = 0
-    int32 limit = 3;
+    repeated string featureId = 3;
+    // [optional] time range filter. If not specified all feature with any timestamp will be returned.
+    TimestampRange timeRange = 4;
```

Comment:
This seems a bit misleading. We should remove time range. 

We return the associated timestamps, so we can leave it to the client to decide if they care things are too old.

Reply:
I agree, i'll amend the PR